### DISCUSSION TITLE
Can not access the webpage from phone

### DISCUSSION THREAD

**Discussion by roshanyadav29** [signag/raspi-cam-srv#29] [Q&A]
I have setup the raspi-cam-srv repo and flask server. I can access the page from rpi as well as my laptop. But unfortunately I can't access it using chrome in my phone. Updated to lastest version, still no luck. Info page attached if it helps. Thanks for the awesome work with repo !!
![image](https://github.com/user-attachments/assets/ad711ac0-a815-4fc0-ae5c-f5f26ea9f5d6)

**Comment by signag** [comment]
Hi, thanks for reporting this issue.

I successfully tried with Chrome on my mobile, connected locally through WiFi as well as remotely through VPN, using http protocol.

Which error does your Chrome browser report?    
Is there probably some policy for your browser or your mobile security environment forcing usage of https?

**Comment by roshanyadav29** [comment]
It could be forcing the use of https. Can you give me way to test it out. I checked in the `settings `> `security `> `Always use secure connections` was turned off. I typed http://rpicam:8000 manually and it didn't work. Screenshots from my mobile phone chrome browser :
![rpicam](https://github.com/user-attachments/assets/8338a561-e5d7-4bc1-8c40-c57c13efb349)
and from my laptop typing the same thing:
![image](https://github.com/user-attachments/assets/8ebf5723-bb67-4ed9-9243-a6545a054769)

double checked, I am connected to same wifi on both devices (turned off the data pack, just to be sure it wasn't using that)

**Comment by signag** [comment]
Did you try pinging ```rpicam``` from your phone?

**Reply by roshanyadav29** [reply]
It says cannot find the host when I do `ping rpicam` using an app called connectbot in my phone. To test it I tried `ping ip_of_my_laptop` and it worked. similarly tried `ping rpicam` from my laptop powershell and its working. Not sure why it can't find rpicam from phone even though I am on the same wifi network.

**Comment by signag** [comment]
Then you will have an issue with name resolution in your network.    
Probably the DNS server is not correctly configured in your WiFi router.

When you do a ping from your laptop, you will see the IP address for rpicam.     
Use this instead of 'rpicam' in the URL in Chrome on your mobile phone.

**Reply by roshanyadav29** [reply]
when I ping from my laptop it shows me IPv6 address (I think), starts with fe80. I ssh into the rpi with `ssh username@rpicam` from laptop and do `ifconfig`. I tried that IP address `ip_address:8888` and it works in the phone. You were right with this being an issue with the name. Any clues to how I could solve these? since rpicam:8888 still works in the laptop, I am not sure how to proceed.

**Reply by signag** [reply]
Honestly, I do not really have any proposal other than going through all your network components and check their TCP/IP settings, especially DNS.
For Android, there is a setting for Private DNS, which should be set to automatic.    
For Windows, name resolution could also work without DNS, if there is an entry in the hosts file.

Sorry

**Reply by roshanyadav29** [reply]
Understood, I will update on this If I end up finding a solution. Thanks for helping to pinpoint the problem.
